internal/wasmldr/src/loader/compiled/tls.rs

```diff
@@ -83,6 +95,17 @@ impl Stream {
 
         Ok(Self::new(tcp, tls))
     }
+
+    fn complete_io(&self) -> Result<(), Error> {
+        let (cap, tls) = &mut *self.lck.write().unwrap();
+
+        while tls.wants_read() || tls.wants_write() {
```

Comment:
`complete_io` already does this check internally

Reply:
It only does one read. See the docs.